Source/Gauntlet/Recording/Implementations/MockIssueRecorder.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  MockIssueRecorder.swift
+//
+//  MIT License
+//
+//  Copyright (c) [2023] The Kroger Co. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in all
+//  copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//  SOFTWARE.
+
+import Foundation
+import XCTest
+
+/// A mock implementation of `IssueRecorder` that can be used to test assertions.
+public class MockIssueRecorder: IssueRecorder {
+    /// A list of all issues that have been recorded.
+    public private(set) var recordedFailures: [RecordedFailure]
+
+    /// Creates a `MockIssueRecorder` instance.
+    public init() {
+        recordedFailures = []
+    }
+
+    /// Clears the list of recorded issues.
+    public func clear() {
+        recordedFailures = []
+    }
+
+    /// Records the provided issue. All recorded issues are stored in the `recordedIssues` array.
+    ///
+    /// - Parameter issue: The issue to record.
```

Comment:
Does not match the function

Reply:
Re-reading this bit of code I realized that the types have the wrong names. The protocol was named IssueRecorder, but that doesn't really make sense anymore since this doesn't handle an XCTIssue directly. I renamed this, and all implementations, to `FailureRecorder` instead, which feels like a better fit.